retrieval_service/datastore/providers/postgres.py

```diff
@@ -39,31 +41,35 @@ class Config(BaseModel, datastore.AbstractConfig):
 
 
 class Client(datastore.Client[Config]):
-    __pool: asyncpg.Pool
+    __pg_ds: PostgresDatastore
```

Comment:
nit: you could consider using "postgres.Client" as the "source" class for AlloyDB / Cloud SQL instead of all 3 relying on PostgresDatastore -- this would validate that the source class meets the Client interface

Reply:
That's a great suggestion! Thanks!

Reply:
Could you elaborate?

Assuming you meant that changing the parent class from `datastore.Client[Config]` to `postgres.Client` for the `alloydb.Client` and `cloudsql_postgres.Client`, I ran into some interesting type checking issues as follows:

```
datastore/providers/cloudsql_postgres.py:51: error: Argument 1 of "create" is incompatible with supertype "Client"; supertype defines the argument type as "Config"  [override]
datastore/providers/cloudsql_postgres.py:51: note: This violates the Liskov substitution principle
datastore/providers/cloudsql_postgres.py:51: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
datastore/providers/alloydb.py:51: error: Argument 1 of "create" is incompatible with supertype "Client"; supertype defines the argument type as "Config"  [override]
datastore/providers/alloydb.py:51: note: This violates the Liskov substitution principle
datastore/providers/alloydb.py:51: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

Reply:
Right now you are doing this:

```mermaid
flowchart LR
    A[AlloyDBClient] --composed by--> D[PostgresDatastore]
    C[CloudSQLPGClient] --composed by--> D
    P[PostgresClient] --composed by--> D
```

I'm just suggesting you get rid of one wrapper class so it looks like this:
```mermaid
flowchart LR
    A[AlloyDBClient] --composed by--> P[PostgresClient]
    C[CloudSQLPGClient] --composed by--> P
```

Which you would implement with something like:
```python

    def __init__(self, pgclient: PostgresClient): 
        self.__pgclient = pgclient

    @classmethod
    async def create(cls, config: Config) -> "Client":
         # connection pool gobble gook 
        return cls(PostgresClient(async_engine))
```